autosynth/synthesizer.py

```diff
@@ -96,22 +100,22 @@ def synthesize(self, environ: typing.Mapping[str, str] = None) -> str:
             command = [sys.executable, self.synth_py_path]
 
         logger.info(command)
-        # Use a temporary file to tee the output, so we can see the output line
-        # by line and still return it as a string.
-        with tempfile.NamedTemporaryFile("wt+") as synth_log_file:
-            tee_proc = subprocess.Popen(
-                ["tee", synth_log_file.name], stdin=subprocess.PIPE
-            )
-            # Invoke synth.py.
-            synth_proc = subprocess.run(
-                command + self.extra_args,
-                stderr=subprocess.STDOUT,
-                stdout=tee_proc.stdin,
-                env=(environ or os.environ),
-                universal_newlines=True,
-            )
-            if synth_proc.returncode:
-                logger.error("Synthesis failed")
-                synth_proc.check_returncode()  # Raise an exception.
-            synth_log_file.seek(0)
-            return synth_log_file.read()
+
+        # Ensure the logfile directory exists
+        logfile.parent.mkdir(parents=True, exist_ok=True)
+        # Tee the output into a provided location so we can see the return the final output
+        tee_proc = subprocess.Popen(["tee", logfile], stdin=subprocess.PIPE)
+        # Invoke synth.py.
+        synth_proc = subprocess.run(
+            command + self.extra_args,
+            stderr=subprocess.STDOUT,
+            stdout=tee_proc.stdin,
+            env=(environ or os.environ),
+            universal_newlines=True,
+        )
+        if synth_proc.returncode:
+            logger.error("Synthesis failed")
+            synth_proc.check_returncode()  # Raise an exception.
+
+        with open(logfile, "r") as fp:
+            return fp.read()
```

Comment:
The synthesize() function returns a string, but line 121 returns a byte array.
Open the file with mode "rt" to read and return a string.

Reply:
Opened with "rt". Do I need to cast to a string or does `read()` change return type based on the the file mode opened with?

Reply:
"rt" is sufficient.  read()'s return type varies with the file mode.  :-/